x-pack/plugins/alerting/common/routes/rule/apis/aggregate/schemas/v1.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+export const aggregateRulesRequestBodySchema = schema.object({
+  search: schema.maybe(schema.string()),
+  default_search_operator: schema.oneOf([schema.literal('OR'), schema.literal('AND')], {
+    defaultValue: 'OR',
+  }),
+  search_fields: schema.maybe(schema.arrayOf(schema.string())),
+  has_reference: schema.maybe(
+    // use nullable as maybe is currently broken
+    // in config-schema
+    schema.nullable(
+      schema.object({
+        type: schema.string(),
+        id: schema.string(),
+      })
+    )
+  ),
+  filter: schema.maybe(schema.string()),
+});
+
+export const ruleAggregationFormattedResponse = schema.object({
```

Comment:
Sorry I didn't see this the first time. But this shouldn't be in `alerting/common`, what you have now is the application type. Only http response types should be in `alerting/common` so lets create a type called `aggregateRulesResponseSchema`: Yes it is kind of verbose but that is desirable in this situation. 

```
export const aggregateRulesResponseBodySchema = schema.object({
  rule_execution_status: schema.recordOf(schema.string(), schema.number()),
  rule_last_run_outcome: schema.recordOf(schema.string(), schema.number()),
  ...
});
```

And the corresponding type:

```
# in aggregate/types/v1.ts
export type AggregateRulesResponseBody = TypeOf<typeof aggregateRulesResponseBodySchema>;

export type AggregateRulesResponse = {
  body: AggregateRulesResponseBody
}
```

`RuleAggregationFormattedResponse` (perhaps name it `RuleAggregationFormattedResult`? since response implies HTTP response) as you have it now should go to a file in `alerting/server/application/rule/methods/aggregate/types` and it would be unversioned. It will be the direct return type of the `aggregateRules` rulesClient method.